proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -280,6 +280,11 @@ message WorkflowExecutionInfo {
     // Should be deprecated once the transition history is fully launched
     repeated VersionedTransition previous_transition_history = 100;
     VersionedTransition last_transition_history_break_point = 101;
+
+    // This is a set of child workflows that were initialized after the reset point in the parent workflow.
+    // The children are identified by the key "workflow_type:workflow_id". When the parent starts to make progress after reset, it uses this data to
+    // determine the right start policy to apply to the child.
+    map<string, bool> children_initialized_post_reset_point = 102;
```

Comment:
Is the `bool` here just a placeholder? 
In either cases, I'd create a separate proto message for this in case any new info needs to be stored later.

Reply:
I'm using the boolean indicator to track children that need to be restarted. Once restarted, I'm setting it false (but not deleting it). I plan to use the existence of false value to detect and ignore results from old child.
I agree that having a separate proto object for this is better (even if it contains a single bool field). Will update it.

Reply:
Ah I see. because we only support Restart option in the beginning, so it will always be set to `true` initially.

Reply:
Added `ResetChildInfo` message. It just contains a bool field (`should_terminate_and_start`) for now.